### PR TITLE
Proper integration of PRTF 

### PR DESCRIPTION
**Pull request by Toonggg**
The new  _radial integration function now checks the input array dimension and chooses the correct
number of dimensions.

### CODE REVIEW

File: src/_spimage_utils.py [hunk 1]
```diff
@@ -39,60 +39,50 @@ def _gaussian_smooth_2d1d(I,sm,precision=1.):
...
-    if 'rout' in kwargs: return numpy.array([radii,values])
-    else: return values
+        return 
+        
+    n_dim = len(image.shape)
+    im_dim = image.shape[0] 
+    im_center = image.shape[0]//2
+    num_shells = int((im_dim-im_center)/shell_thickness) 
+    
+    if n_dim == 2:
```
Comment (FilipeMaia) [inline src/_spimage_utils.py:59]: We should avoid have almost equal different code paths for 2 and 3 dims.

File: src/_spimage_utils.py [hunk 2]
```diff
@@ -39,60 +39,50 @@ def _gaussian_smooth_2d1d(I,sm,precision=1.):
...
+        r_out = numpy.arange(shell_thickness, (num_shells+1) * shell_thickness, shell_thickness) 
+        r_in = r_out - 1 
+        radial_mask = ((x - c[0]) ** 2 + (y - c[1]) ** 2  >= r_in[:, None, None] ** 2) & ((x - c[0]) ** 2 + (y - c[1]) ** 2 < r_out[:, None, None] ** 2)   
+    elif n_dim == 3:
+        x, y, z = numpy.meshgrid(numpy.arange(image.shape[0]), numpy.arange(image.shape[1]), numpy.arange(image.shape[2]), indexing='ij')
+        c = numpy.array([im_center, im_center, im_center])
+        
+        r_out = numpy.arange(shell_thickness, (num_shells+1) * shell_thickness, shell_thickness) 
+        r_in = r_out - 1 
+        radial_mask = ((x - c[0]) ** 2 + (y - c[1]) ** 2 + (z - c[2]) ** 2 >= r_in[:, None, None, None] ** 2) & ((x - c[0]) ** 2 + (y - c[1]) ** 2 + (z - c[2]) ** 2 < r_out[:, None, None, None] ** 2)
```
Comment (FilipeMaia) [inline src/_spimage_utils.py:72]: This is too complicated for a single line. Split it.

File: src/_spimage_utils.py [hunk 3]
```diff
@@ -39,60 +39,50 @@ def _gaussian_smooth_2d1d(I,sm,precision=1.):
...
+
+    prtf_r = numpy.array([f(image[shell]) for shell in radial_mask]) 
+    return prtf_r[numpy.isfinite(prtf_r)] 
+def radial_mean(image, **kwargs): 
+    return _radial(image,"mean",**kwargs)
+def radial_std(image, **kwargs):
+    return _radial(image, "std",**kwargs) 
 def radial_sum(image, **kwargs):
     return _radial(image,"sum",**kwargs)
-def radial_std(image, **kwargs):
```
Comment (FilipeMaia) [inline src/_spimage_utils.py:89]: Is the radial std and median used anywhere?

Comment (Toonggg) [reply src/_spimage_utils.py:89]: No. These were kept to stay consistent with the "older" function. Maybe its better to have the mean as a default, but if someone would like to use a different function f, they can supply their own. That also cleans up the first part of the code.

File: src/_spimage_utils.py (hunk 2 above)
Comment (Toonggg) [reply src/_spimage_utils.py:72]: For this, I was thinking to have the entire indexing surrounded by parentheses and then split for the >= r_in and r_out in two lines.

**Pull request closed without merging**